### DISCUSSION TITLE
Reverse Test

### DISCUSSION THREAD

**Discussion by jb-home** [thkl/hap-homematic#341] [Ideas]
Hallo @thkl,

wie wäre es mit einem Test, ob die Geräte die in HAP-Homematic konfiguriert sind auch alle noch in der CCU registriert sind?

Manchmal baut man ja Steckdosen etc um und lernt diese ganz ordentlich in der CCU ab. Vergisst aber, diese aus HAP zu entfernen. Und später fällt es dann einem wieder ein oder auf.

Denke du könntest die bei HAP registrierten Geräte (hast ja die Adresse) mit den in der CCU verfügbaren Geräten abgleichen.
Denke die Sondergeräte müsste man nicht unbedingt testen, um den Aufwand gering zu halten.

**Comment by thkl** [comment]
Moinsen, ja das ist keine so dumme Idee. Ich kann Rega fragen, ob ein Datenpunkt existiert. Ich würde aber ungern den Kram automatisch löschen. Mir würde ein Popup beim Start des WebUI besser gefallen, was einen die "toten" Geräte anzeigt und man kann die dann mit einem Klick löschen ...

**Reply by jb-home** [reply]
Automatisch löschen fände ich auch nicht gut.

Denke ggf. so ein Menüpunkt wie bei HVL mit den gelöschten Geräten.

Automatisch könntest du teilweise so umsetzen, dass du bei den Geräten die in HAP sind aber nicht mehr in der CCU, das UNREACHABLE setzt, dann könnte man das auch in HomeKit sehen.

**Reply by thkl** [reply]
![image](https://user-images.githubusercontent.com/884355/107809517-8679f580-6d6b-11eb-8af7-c70bed5dbb1f.png)

**Reply by jb-home** [reply]
Sehr geil!
Freu mich drauf... Auch wenn bei mir im Moment hoffentlich alles in Ordnung ist ;)

**Reply by jb-home** [reply]
Hallo Thomas,

ich hatte zum Glück doch noch ein paar tote Geräte, wodurch ich die neue Erweiterung testen konnte.
Wenn ich bei einem auf löschen klicke, scheint der gesamte test erneut zu laufen, was etwas langsam erscheint. Ist aber OK und wahrscheinlich auch am gründlichsten.

Nachdem ich jedoch den letzen gelöscht habe, blieb die Oberfläche so grau. Da half dann nur ein reload der Seite.

**Reply by thkl** [reply]
Oh, ok. Probier ich noch mal aus.  

Und ja das mit dem Erneuten Laden ist so gewollt, weil am Einfachsten 🙃
